src/Mvc/Mvc.RazorPages/src/ApplicationModels/PageConventionCollection.cs

```diff
@@ -14,6 +14,7 @@ public class PageConventionCollection : Collection<IPageConvention>
         /// Initializes a new instance of the <see cref="PageConventionCollection"/> class that is empty.
         /// </summary>
         public PageConventionCollection()
+            : this((MvcOptions)null)
```

Comment:
Would it make things better/easier if this was `IServiceProvider` instead? I'm worried about cycles in options, not sure if `IServiceProvider` makes that better or now.

Reply:
We could lazy load the options when you do a convention. Conventions get applied much later during execution, so things are definitely wired up by then